tensorboard/webapp/metrics/views/card_renderer/utils.ts

```diff
@@ -91,50 +91,32 @@ export interface TimeSelectionView {
   clipped: boolean;
 }
 
+function boundNumber(value: number, min: number, max: number) {
+  if (value < min) {
+    return min;
+  }
+  if (value > max) {
+    return max;
+  }
+  return value;
+}
+
 export function maybeClipLinkedTimeSelection(
   timeSelection: TimeSelection,
   minStep: number,
   maxStep: number
 ): TimeSelectionView {
-  if (
-    // Case when timeSelection contains extents.
-    (timeSelection.start.step <= minStep &&
-      timeSelection.end &&
-      maxStep <= timeSelection.end.step) ||
-    // Case when start of timeSelection is within extent.
-    (minStep <= timeSelection.start.step &&
-      timeSelection.start.step <= maxStep) ||
-    // Case when end of timeSelection is within extent.
-    (timeSelection.end &&
-      minStep <= timeSelection.end?.step &&
-      timeSelection.end?.step <= maxStep)
-  ) {
-    return {
-      startStep: timeSelection.start.step,
-      endStep: timeSelection.end?.step ?? null,
-      clipped: false,
-    };
-  }
-
-  // When time selection and data extent (in step axis) do not overlap,
-  // default single select min or max data extent depending on which side
-  // the time selection is at.
-
-  // Case when time selection is on the right of the maximum of the
-  // time series.
-  if (maxStep <= timeSelection.start.step) {
-    return {
-      startStep: maxStep,
-      endStep: null,
-      clipped: true,
-    };
-  }
-  // Case when time selection is on the left of the minimum of the time
-  // series.
+  const startStep = boundNumber(timeSelection.start.step, minStep, maxStep);
```

Comment:
As offline discussion, propose some name changes:

const startStep = timeSelection.start.step;
const maybeClippedStartStep = clipStepWithinMinMax(startStep, minStep, maxStep)
...
const clipped = maybeClippedStartStep !== startStep || ...
    
What do you think?